### ISSUE TITLE
Incorrect escaping inside STYLE and SCRIPT elements.

### ISSUE THREAD

**Issue by mbostock** [observablehq/htl#6]
STYLE and SCRIPT elements have special behavior regarding escaping: they use “RAWTEXT” rather than “DATA” mode. For example, this is not an ampersand character reference:

```js
html`<script>hello&amp;</script>`
```

I think this means we’ll need to track the element name when we enter the DATA state, and if it’s STYLE or SCRIPT, enter the RAWTEXT state instead. And then likewise we’ll have to handle the “appropriate end tag” to determine when we exit the RAWTEXT state.

**Comment by mbostock** [comment]
As another example, the ampersand should not be encoded as `&#38;` here:

```js
html`<style>

p {
  background-image: url(${"foo.png?bar=1&baz=2"});
}

</style>`
```

**Comment by clarkevans** [comment]
We implemented this within a [fork](https://github.com/clarkevans/HypertextLiteral.jl) it turned out to be relatively easy.  Most of the state machine already tracks what you need to keep the current element name, and you only have to match on `style`/`script` tags (case independently).  Note that besides not permitting `<style>` or `<script>` literal values, additionally `<script>` doesn't like comments `<!--`.  There's a design choice to try and save the user from injecting these values, or, letting them do it (since including `<script>` within Javascript string is a gotya that you can find on stackoverflow).  At this time, our fork is trying to detect this error, but there's a speed cost, so we may rip it out.

**Comment by mbostock** [comment]
@clarkevans Thanks for the pointer! I’d like to fix this together with #18; see https://github.com/observablehq/htl/pull/21/files/c537630c5bb56ad2be18540dc3071bf409429933..d90834ff342f96d66f70cfde0644a124dd8f44d5#r554461977.
